canonical/lxd-ui
# feat: add support for instance/profile host path devices [WD-17682]

## Done

- Adjusted devices table in instance/profile overview page to show host path disk devices
- Added support for create/edit host path disk devices in instance/profile configuration pages
- Adjusted the workflow for adding disk devices, now it is a multi-step flow, starting with selecting the type of disk device (volume or host path).
- Adjusted/added e2e tests to reflect the above changes.

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - Create a new profile, add both custom volume and host path disk devices to it.
    - Create an instance and apply the profile to that instance, make sure all disk devices are inherited.
    - Create custom volume and host path disk devices for the above instance, using different volume and mount point.
    - Start the instance, go to its terminal and make sure that path corresponding to every disk device exists. Make sure that the mounted host path has the correct content (should be the same to the path on your machine).
    - Go to the instance/profile overview page, check the device list table and make sure disk devices are displayed correctly.
    - Try modifying the volume and host path disk devices on the profile and instance.
